airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -106,6 +106,15 @@ SELECT pg_relation_filenode('%s')
       WITH block_sz AS (SELECT current_setting('block_size')::int), rel_sz AS (select pg_relation_size('%s')) SELECT * from block_sz, rel_sz
       """;
 
+  /**
+   * Query estimates the max tuple in a page. We are estimating in two ways and selecting the greatest
+   * value.
+   */
+  public static final String CTID_ESTIMATE_MAX_TUPLE =
+      """
+      SELECT COALESCE(MAX((ctid::text::point)[1]::int), 0) AS max_tuple FROM "%s"."%s"
```

Comment:
just checking that `int` is large enough and we don't need a bigint or long here?

Reply:
a page is 8K typically, so it won't be able to get anywhere close to MAX_INT